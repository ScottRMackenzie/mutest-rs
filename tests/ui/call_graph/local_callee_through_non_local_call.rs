//@ print-targets
//@ stdout
//@ stderr: empty
//@ aux-build: crate_with_calls.rs

extern crate crate_with_calls;

use std::iter::{self, FromIterator};

struct Collectable;

fn from_iter_impl() {}
impl FromIterator<()> for Collectable {
    fn from_iter<T>(_iter: T) -> Self {
        from_iter_impl();
        Collectable
    }
}

#[test]
fn test_local_callee_through_std() {
    let _ = iter::repeat(()).take(3).collect::<Collectable>();
}

struct ImplsExternTrait;

fn extern_trait_fn_impl() {}
impl crate_with_calls::ExternTrait for ImplsExternTrait {
    fn extern_trait_fn() {
        extern_trait_fn_impl();
    }
}

#[test]
fn test_local_callee_through_extern_crate() {
    crate_with_calls::extern_fn_calling_trait_fn::<ImplsExternTrait>();
}
