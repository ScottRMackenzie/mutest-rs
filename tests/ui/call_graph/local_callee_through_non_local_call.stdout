tests -(2)-> from_iter_impl at tests/ui/call_graph/local_callee_through_non_local_call.rs:12:1: 12:20 (#0)
  (2) test_local_callee_through_std

tests -(1)-> <Collectable as std::iter::FromIterator<()>>::from_iter at tests/ui/call_graph/local_callee_through_non_local_call.rs:14:5: 14:38 (#0)
  (1) test_local_callee_through_std

tests -(2)-> extern_trait_fn_impl at tests/ui/call_graph/local_callee_through_non_local_call.rs:27:1: 27:26 (#0)
  (2) test_local_callee_through_extern_crate

tests -(1)-> <ImplsExternTrait as crate_with_calls::ExternTrait>::extern_trait_fn at tests/ui/call_graph/local_callee_through_non_local_call.rs:29:5: 29:25 (#0)
  (1) test_local_callee_through_extern_crate

targets: 4 total; 4 safe; 0 unsafe (0 tainted)
